resolwe_api/resolwe.py

```diff
@@ -493,3 +337,102 @@ def __call__(self, request):
         # if r.path_url != '/upload/':
         #     r.headers['X-SubscribeID'] = self.subscribe_id
         return request
+
+
+class ResolweQuerry(object):
+    """
+    Enable querries on data, collection and sample endpoints in Resolwe.
+
+    Each Resolwe instance (for example "re") has threee "endpoints": re.data,
+    re.collections and re.sample. Each such andpooint is an instance of
+    ResolweQuerry class. ResolweQuerry enables querries on
+    corresponding objects, for example:
+
+    re.data.get(42) # return Data object with ID 42.
+    re.sample.filter(contributor=1) # return all samples made by contributor 1
+
+    Detailed description of methods can be found in method docstrings.
+    """
+
+    def __init__(self, resolwe, type_):
+        self.resolwe = resolwe
+        self.object = type_
+        self.type = type_.endpoint
+        # Make separate sluber API instance, since the one from
+        # Resolwe colud not be used as desired.
+        self.connection = slumber.API(self.resolwe.url,
```

Comment:
Only option I see to reuse connection is to make somthing like this:

def __init__(self, resolwe, type):
    ...
    self.api = None
    if type == Data:
        self.api = resolwe.api.data
    if type == Sample:
        self.api = resolwe.api.sample
    if type == Collection:
        self.api = resolwe.api.collection
    ...

Which i thought was ugly. But I guess making another connection is far more uglier. :-)
And since we don't expect to have more types of **type** parameter (only study at sight) it seems like an OK  solution. Any other idea?